remote/rest_util_test.go

```diff
@@ -37,38 +36,45 @@ tryStartService:
 				continue tryStartService
 			}
 
+			srvCtx, srvCancel := context.WithCancel(ctx)
 			go func() {
```

Comment:
would it work to combine the two goroutines? Something like

```
failedToConnect := make(chan struct{})
go func() {
    srvCtx, srvCancel := context.WithCancel(ctx)
    defer srvCancel()
    grip.Warning(app.Run(srvCtx))

    select {
    case <-ctx.Done():
    case <-failedToConnect:
    }
```

Reply:
We can't do this, because `Run()` is a blocking operation, which would prevent the select from stopping the application when the client fails to connect.